reference-adapter/mgramseva-ifix-adapter/src/main/java/org/egov/ifix/service/impl/PspclEventServiceImpl.java

```diff
@@ -0,0 +1,251 @@
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.gson.Gson;
+import lombok.extern.slf4j.Slf4j;
+import org.egov.ifix.exception.GenericCustomException;
+import org.egov.ifix.models.fiscalEvent.FiscalEvent;
+import org.egov.ifix.models.fiscalEvent.FiscalEventResponseDTO;
+import org.egov.ifix.models.mdms.Tenant;
+import org.egov.ifix.models.mgramseva.ChallanResponseDTO;
+import org.egov.ifix.models.mgramseva.SearchChallanResponseDTO;
+import org.egov.ifix.repository.FiscalEventRepository;
+import org.egov.ifix.repository.MdmsRepository;
+import org.egov.ifix.repository.MgramsevaChallanRepository;
+import org.egov.ifix.repository.MgramsevaVendorRepository;
+import org.egov.ifix.service.*;
+import org.egov.ifix.utils.ApplicationConfiguration;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+
+import javax.validation.constraints.NotNull;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+
+import static org.egov.ifix.utils.EventConstants.*;
+
+@Slf4j
+@Service
+public class PspclEventServiceImpl implements PspclEventService {
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private ApplicationConfiguration applicationConfiguration;
+
+    @Autowired
+    private FiscalEventRepository fiscalEventRepository;
+
+    @Autowired
+    private MdmsRepository mdmsRepository;
+
+    @Autowired
+    private MgramsevaChallanRepository mgramsevaChallanRepository;
+
+    @Autowired
+    private AuthTokenService authTokenService;
+
+    @Autowired
+    private EventPostingDetailService eventPostingDetailService;
+
+    @Autowired
+    private PspclEventPersistenceService pspclEventPersistenceService;
+
+    @Autowired
+    private MgramsevaVendorRepository mgramsevaVendorRepository;
+
+    @Autowired
+    private CollectionService collectionService;
+
+    @Autowired
+    private MgramsevaChallanService mgramsevaChallanService;
+
+    /**
+     * @param eventType
+     */
+    @Override
+    public void processPspclEventForMgramseva(String eventType) {
+        FiscalEventResponseDTO fiscalEventResponseDTO = fiscalEventRepository
+                .collectFiscalEvent(applicationConfiguration.getPspclIfixEventReceiverName(), eventType);
+
+        if (fiscalEventResponseDTO == null || fiscalEventResponseDTO.getFiscalEvent() == null
+                || fiscalEventResponseDTO.getFiscalEvent().isEmpty()) {
+            pspclEventPersistenceService.saveFailedPspclEventDetail(NA, eventType, NA,
+                    "Unable to get any PSPCL event from fiscal event (iFix Core)");
+            throw new GenericCustomException(PSPCL, "Unable to get any PSPCL event from fiscal event (iFix Core)");
```

Comment:
This isn't an exception. There will be many times when there are no new fiscal events(say, in last 48 hours). So just log the message. But don't throw it as an error.